src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -189,19 +189,27 @@ def train_job(train_cfg, train_dmatrix, val_dmatrix, train_val_dmatrix, model_di
     :param is_master: True if single node training, or the current node is the master node in distributed training.
     """
     # Parse arguments for train() API
-    early_stopping_rounds = train_cfg.pop('early_stopping_rounds', None)
     num_round = train_cfg.pop("num_round")
 
     # Evaluation metrics to use with train() API
     tuning_objective_metric_param = train_cfg.pop("_tuning_objective_metric", None)
     eval_metric = train_cfg.get("eval_metric")
-    cleaned_eval_metric, configured_feval = train_utils.get_eval_metrics_and_feval(
+    cleaned_eval_metric, configured_feval, tuning_objective_metric = train_utils.get_eval_metrics_and_feval(
         tuning_objective_metric_param, eval_metric)
     if cleaned_eval_metric:
         train_cfg['eval_metric'] = cleaned_eval_metric
     else:
         train_cfg.pop('eval_metric', None)
 
+    early_stopping_rounds = train_cfg.pop('early_stopping_rounds', None)
+    early_stopping_data = 'validation' if val_dmatrix else 'train'
```

Comment:
Hi Haifeng, see this line for `early_stopping_data`.

Reply:
fixed